Example/Tests/CISAdAnalyticsTestDouble.swift

```diff
@@ -9,48 +9,88 @@
 import ConvivaSDK
 import Foundation
 
-class CISAdAnalyticsTestDouble: NSObject, CISAdAnalyticsProtocol {
+class CISAdAnalyticsTestDouble: NSObject, CISAdAnalyticsProtocol, TestDoubleDataSource {
     func isAirPlaying() -> Bool {
         false
     }
 
     func setAdInfo(_ adInfo: [AnyHashable: Any]) {
+        var args = [String: String]()
+        adInfo.forEach { key, value in args[String(describing: key)] = String(describing: value) }
+        spy(functionName: "setAdInfo", args: args)
     }
 
     func setAdPlayerInfo(_ adPlayerInfo: [AnyHashable: Any]) {
+        var args = [String: String]()
+        adPlayerInfo.forEach { key, value in args[String(describing: key)] = String(describing: value) }
+        spy(functionName: "setAdPlayerInfo", args: args)
     }
 
     func reportAdFailed(_ errorMessage: String, adInfo: [AnyHashable: Any]?) {
+        var args = [String: String]()
+        args["errorMessage"] = errorMessage
+        adInfo?.forEach { key, value in args[String(describing: key)] = String(describing: value) }
+        spy(functionName: "reportAdFailed", args: args)
     }
 
     func reportAdLoaded(_ adInfo: [AnyHashable: Any]?) {
+        var args = [String: String]()
+        adInfo?.forEach { key, value in args[String(describing: key)] = String(describing: value) }
+        spy(functionName: "reportAdLoaded", args: args)
     }
 
     func reportAdStarted(_ adInfo: [AnyHashable: Any]?) {
+        var args = [String: String]()
+        adInfo?.forEach { key, value in args[String(describing: key)] = String(describing: value) }
+        spy(functionName: "reportAdStarted", args: args)
     }
 
     func reportAdEnded() {
+        spy(functionName: "reportAdEnded")
     }
 
     func reportAdError(_ errorMessage: String, severity: ErrorSeverity) {
+        spy(functionName: "reportAdError", args: [
+            "errorMessage": errorMessage,
+            "severity": "\(severity)"
+        ])
     }
 
     func reportAdSkipped() {
+        spy(functionName: "reportAdSkipped")
     }
 
     func reportAdPlayerEvent(_ eventType: String, details: [AnyHashable: Any]?) {
+        var args = [String: String]()
+        args["eventType"] = eventType
+        details?.forEach { key, value in args[String(describing: key)] = String(describing: value) }
+        spy(functionName: "reportAdPlayerEvent", args: args)
     }
 
     func reportAdMetric(_ key: String, value: Any) {
+        spy(functionName: "reportAdMetric", args: [
+            "key": key,
+            "value": "\(value)"
+        ])
     }
 
     func setContentSessionID(_ sessionID: Int32) {
+        spy(functionName: "setContentSessionID", args: [
+            "sessionID": "\(sessionID)"
+        ])
     }
 
     func setAdListener(_ adProxy: Any?, andInfo info: [AnyHashable: Any]) {
+        var args = [String: String]()
+        args["adProxy"] = "\(adProxy ?? "nil")"
```

Comment:
Why do you use `nil` as String here and not as Optional 🤔

Reply:
Not very fit in swift :D 

How would that look like?
The `spy` at the end of the function expects a `String String` mapping.

Reply:
https://github.com/bitmovin/bitmovin-player-ios-analytics-conviva/pull/69/commits/fc9d160b3a1c4dc670a570e77eecf656c3c06335